cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocJarUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteStreams;
+import joptsimple.OptionSpec;
+import org.apache.twill.api.ClassAcceptor;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+import org.apache.twill.internal.ApplicationBundler;
+import org.apache.twill.internal.appmaster.ApplicationMasterMain;
+import org.apache.twill.internal.container.TwillContainerMain;
+import org.apache.twill.internal.utils.Dependencies;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+
+/**
+ * Util class to build jar files needed by {@code DataprocRuntimeJobManager}.
+ */
+final class DataprocJarUtil {
+
+  /**
+   * Returns twill bundle jar.
+   *
+   * @param locationFactory location factory to create location
+   * @return a runtime jar file
+   * @throws IOException any error while building the jar
+   */
+  static RuntimeLocalFile getTwillJar(LocationFactory locationFactory) throws IOException {
+    ApplicationBundler bundler = new ApplicationBundler(new ClassAcceptor() {
+      @Override
+      public boolean accept(String className, URL classUrl, URL classPathUrl) {
+        return !className.startsWith("org.apache.hadoop") && !classPathUrl.toString().contains("spark-assembly");
```

Comment:
Similar filtering is also being used in MapReduceRuntimeService
https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java#L782

Reply:
This is the spark-assembly class that had conflict. 
```
Caused by: java.lang.NoSuchMethodError: org.apache.hadoop.http.HttpConfig.isSecure()Z
	at org.apache.hadoop.yarn.conf.YarnConfiguration.<clinit>(YarnConfiguration.java:322)
	at io.cdap.cdap.runtime.spi.runtimejob.DataprocRuntimeEnvironment.initialize(DataprocRuntimeEnvironment.java:87)
	... 10 more

Since we already have a workaround, I opened a jira to investigate this further. 
https://issues.cask.co/browse/CDAP-16433